core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestAnalyzeAction.java

```diff
@@ -67,91 +64,92 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
 
         AnalyzeRequest analyzeRequest = new AnalyzeRequest(request.param("index"));
 
-        buildFromContent(RestActions.getRestContent(request), analyzeRequest, parseFieldMatcher);
+        try (XContentParser parser = request.contentOrSourceParamParser()) {
+            buildFromContent(parser, analyzeRequest, parseFieldMatcher);
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Failed to parse request body", e);
+        }
 
         return channel -> client.admin().indices().analyze(analyzeRequest, new RestToXContentListener<>(channel));
     }
 
-    static void buildFromContent(BytesReference content, AnalyzeRequest analyzeRequest, ParseFieldMatcher parseFieldMatcher) {
-        try (XContentParser parser = XContentHelper.createParser(content)) {
```

Comment:
I moved the `try` into the calling method so this is mostly indentation changing.